DataAccessGeneration/Generator.cs

```diff
@@ -352,8 +352,11 @@ private ResultMetaData GetResultMetaData(ProcedureSetting procedureSetting, List
     {
         var userDefinedTypeNames = userDefinedTypes.Select(x => x.TableTypeName).Distinct().ToList();
         var parameters = lookup.GetParametersForProcedure(settings.SchemaName, procedureSetting.Proc);
-        var resultColumns = lookup.GetResultDefinitionsForProcedures(settings.SchemaName, procedureSetting.Proc, parameters,
-            allowProcedureExecution: procedureSetting.Return != ReturnType.None);
+        var resultColumns =
+            procedureSetting.LookupOutputTypes
+                ? lookup.GetResultDefinitionsForProcedures(settings.SchemaName, procedureSetting.Proc, parameters,
+                    executeDuringGeneration: procedureSetting.Return != ReturnType.None && procedureSetting.Return != ReturnType.Output)
```

Comment:
this can be replaced with 
``executeDuringGeneration: procedureSettings.LookupOutputTypes``